Fixes for slovak language-specific keycodes

KraXen72 opened the pull request:
## Description
i tried to use the slovak keymap, but ran into a few issues:
- two dead accent characters were switched
- SK_RNGA was defined twice, in 2 different ways, but causing an error when compiling
- the commented line may be removed, but i don't know which version of SK_RNGA you guys prefer, so i kept it commented
- i tested this modified keymap on my kyria for a week and it works as it should
  
## Types of Changes
- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation
  
## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

fauxpark commented on quantum/keymap_extras/keymap_slovak.h:
```diff
@@ -141,9 +141,9 @@
-#define SK_CIRC ALGR(SK_LCAR) // ^ (dead)
-#define SK_BREV ALGR(SK_SCAR) // ˘ (dead)
-#define SK_RNGA ALGR(SK_CCAR) // ° (dead)
+#define SK_BREV ALGR(SK_LCAR) // ˘ (dead)
+#define SK_CIRC ALGR(SK_SCAR) // ^ (dead)
+//#define SK_RNGA ALGR(SK_CCAR) // ° (dead)
```
```suggestion
#define SK_CIRC ALGR(SK_SCAR) // ^ (dead)
#define SK_BREV ALGR(SK_CCAR) // ˘ (dead)
```
Per the Windows OSK:
![image](https://user-images.githubusercontent.com/4781841/150750471-72dd1748-6e09-44c8-aaf9-054ebe4dba2b.png)

KraXen72 replied:
ok. how can i accept the change you proposed?

precondition replied:
@KraXen72 's changes are correct for the Slovak layout on Linux (https://i.imgur.com/moFZm0v.png) and it is also correct if we look at the Slovak keyboard layout shown in the official Microsoft docs: https://docs.microsoft.com/en-us/globalization/keyboards/kbdsl.html (Click on the <kbd>AltGr</kbd> key in that last website to activate the AltGr "layer"). [This](http://kbdlayout.info/KBDSL/) also confirms.

KraXen72 replied:
actually, I'll have to check if this would work, i think the sk keymap was updated at some point

precondition replied:
That seems correct according to the Linux Slovak keyboard (https://i.imgur.com/moFZm0v.png) and according to the official Microsoft docs : https://docs.microsoft.com/en-us/globalization/keyboards/kbdsl.html (Press the <kbd>AltGr</kbd> key on this site to visualise the AltGr "layer").  [This other site](http://kbdlayout.info/KBDSL/) also confirms that <kbd>AltGr</kbd>+<kbd>š</kbd> = <kbd>^</kbd> and <kbd>AltGr</kbd>+<kbd>č</kbd> = <kbd>˘</kbd>

fauxpark replied:
If the Linux or macOS leyouts differ from Windows then they would need to be added as their own headers - refer to the other files in `keymap_extras`.

KraXen72 replied:
just checked it on my machine and you are indeed correct. i thought you changed the two keys i fixed but you found another bug in the slovak keymap, and it should be the way you suggested

KraXen72 commented:
i added placeholders, so the whole deifining isn't shifted and doesen't consufe people trying to verify it.
also added a missing character from the keymapviz

KraXen72 commented:
alternatively, you could rename the two charactes that can be typed in different ways, so people can choose which one they want. i recently had just this use case where i had a key override for the shifted way to achieve this, so i wanted the alrgr way

KraXen72 commented:
added the keys as alternatives, feel free to check my naming if it's good, just went off of what sounded okay

fauxpark commented on quantum/keymap_extras/keymap_slovak.h:
```diff
@@ -128,7 +128,7 @@
 /* AltGr symbols
  * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
- * │   │ ~ │   │ ^ │ ˘ │ ° │ ˛ │ ` │ ˙ │   │ ˝ │ ¨ │ ¸ │       │
+ * │   │ ~ │ ˇ │ ^ │ ˘ │ ° │ ˛ │ ` │ ˙ │   │ ˝ │ ¨ │ ¸ │       │
```
```suggestion
 * │   │ ~ │   │ ^ │ ˘ │ ° │ ˛ │ ` │ ˙ │   │ ˝ │ ¨ │ ¸ │       │
```

KraXen72 replied:
well whatever i don't really care

fauxpark commented on quantum/keymap_extras/keymap_slovak.h:
```diff
@@ -141,9 +141,10 @@
-#define SK_CIRC ALGR(SK_LCAR) // ^ (dead)
-#define SK_BREV ALGR(SK_SCAR) // ˘ (dead)
-#define SK_RNGA ALGR(SK_CCAR) // ° (dead)
+#define SK_CNAL ALGR(SK_LCAR) //ˇ (dead) as in caron alternative
+#define SK_CIRC ALGR(SK_SCAR) // ^ (dead)
+#define SK_BREV ALGR(SK_CCAR) // ˘ (dead)
+#define SK_DGAL ALGR(SK_TCAR) // ° (dead) as in degree alternative
```
```suggestion
#define SK_CIRC ALGR(SK_SCAR) // ^ (dead)
#define SK_BREV ALGR(SK_CCAR) // ˘ (dead)
```
These are not needed as they're already defined above.

KraXen72 replied:
yes, but for example, if i set up a key override for the normal caron key to something else, and i still want to be able to type the caron, i can do it this way. since this is already in the os layout, i don't see a point in not telling the user you can also get the caron like this. that's why i added the names ending as AL, as in ALTGR or ALTERNATIVE. would this be a problem? if yes, why?

fauxpark replied:
This is the convention I've gone with. If a character is available in a "less-modified" way, we favour that one and ignore the other(s) for simplicity. It is not that common anyway. The exception is when the deadness of the key differs.

KraXen72 replied:
well, okay.
